homeassistant/components/mazda/climate.py

```diff
@@ -0,0 +1,186 @@
+"""Platform for Mazda climate integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pymazda import Client as MazdaAPIClient
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.temperature import convert as convert_temperature
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DATA_REGION, DOMAIN
+
+PRESET_DEFROSTER_OFF = "Defroster Off"
+PRESET_DEFROSTER_FRONT = "Front Defroster"
+PRESET_DEFROSTER_REAR = "Rear Defroster"
+PRESET_DEFROSTER_FRONT_AND_REAR = "Front and Rear Defroster"
+
+
+def _front_defroster_enabled(preset_mode: str | None) -> bool:
```

Comment:
Can `preset_mode` actually be `None`? It's going to fail in this case. I suggest you either change the signature, or return `False` when `preset_mode` is `None`.

Reply:
I added the `| None` because the type of `self._attr_preset_mode` is `str | None`, so the type checker would complain on lines 170 and 171 if I didn't add it. Realistically, this shouldn't ever happen, because we assign a value to `self._attr_preset_mode` in `_update_state_attributes()` which is called from the constructor.

Reply:
@OnFreund and @bdr99 any updates on this PR please? It’s been outstanding for a while now

Reply:
I believe I have completed everything that was required from my side.

@OnFreund Are there any other changes needed in order to get your approval on this PR?

@bdraco Also pinging you, since you previously reviewed this PR.

Reply:
Please don't ping. Please re-request review instead.

Reply:
Hi all. Sorry to chase again. Any progress with this? I have a few automation ideas that need these additional entities to facilitate switching a PodPoint on and off depending on climate state